core/src/main/kotlin/in/specmatic/core/utilities/ContractSource.kt

```diff
@@ -22,11 +19,12 @@ sealed class ContractSource {
 
 data class GitRepo(
     val gitRepositoryURL: String,
+    val gitBranch: String?,
     override val testContracts: List<String>,
     override val stubContracts: List<String>
 ) : ContractSource() {
     val repoName = gitRepositoryURL.split("/").last().removeSuffix(".git")
-
+    val branchName = gitBranch
```

Comment:
Can we not directly use gitBranch? just curious why we need to declare another value here.